internal/client/ssh.go

```diff
@@ -0,0 +1,349 @@
+		}
+		if needsInstallAgent {
+			packagesToInstall += " bluechi-agent"
+		}
+		if packagesToInstall == "" {
+			return nil
+		}
+
+		output, err := c.runCommand(fmt.Sprintf("dnf install -y %s", packagesToInstall))
+		if err != nil {
+			return fmt.Errorf("failed to install packages '%s': %s", packagesToInstall, output)
+		}
+	}
+
+	return nil
+}
+
+func (c *SSHClient) CreateControllerConfig(file string, cfg BlueChiControllerConfig) error {
+	cmd := fmt.Sprintf("bash -c 'echo \"%s\" > %s'", cfg.Serialize(), path.Join(BlueChiControllerConfdDirectory, file))
+	output, err := c.runCommand(cmd)
+	if err != nil {
+		return fmt.Errorf("failed to create controller config file: %s", string(output))
+	}
+
+	return nil
+}
+
+func (c *SSHClient) RemoveControllerConfig(file string) error {
+	cmd := fmt.Sprintf("rm -f %s", path.Join(BlueChiControllerConfdDirectory, file))
+	output, err := c.runCommand(cmd)
+	if err != nil {
+		return fmt.Errorf("failed to remove controller config file: %s", string(output))
+	}
+
+	return nil
+}
+
+func (c *SSHClient) RestartBlueChiController() error {
+	output, err := c.runCommand("systemctl start bluechi-controller")
+	if err != nil {
+		return fmt.Errorf("failed to restart controller service: %s", string(output))
+	}
+
+	return nil
+}
+
+func (c *SSHClient) StopBlueChiController() error {
+	output, err := c.runCommand("systemctl stop bluechi-controller")
+	if err != nil {
+		return fmt.Errorf("failed to stop controller service: %s", string(output))
+	}
+
+	return nil
+}
+
+func (c *SSHClient) CreateAgentConfig(file string, cfg BlueChiAgentConfig) error {
+	cmd := fmt.Sprintf("bash -c 'echo \"%s\" > %s'", cfg.Serialize(), path.Join(BlueChiAgentConfdDirectory, file))
+	output, err := c.runCommand(cmd)
+	if err != nil {
+		return fmt.Errorf("failed to create agent config file: %s", string(output))
+	}
+
+	return nil
+}
+
+func (c *SSHClient) RemoveAgentConfig(file string) error {
+	cmd := fmt.Sprintf("rm -f %s", path.Join(BlueChiAgentConfdDirectory, file))
+	output, err := c.runCommand(cmd)
+	if err != nil {
+		return fmt.Errorf("failed to remove agent config file: %s", string(output))
+	}
+
+	return nil
+}
+
+func (c *SSHClient) RestartBlueChiAgent() error {
+	output, err := c.runCommand("systemctl start bluechi-agent")
+	if err != nil {
+		return fmt.Errorf("failed to restart agent service: %s", string(output))
+	}
+
+	return nil
+}
+
+func (c *SSHClient) StopBlueChiAgent() error {
+	output, err := c.runCommand("systemctl stop bluechi-agent")
+	if err != nil {
+		return fmt.Errorf("failed to stop agent service: %s", string(output))
+	}
+
+	return nil
+}
+
+func NewSSHClient(host string, user string, password string, pkPath string, insecureIgnoreHostKey bool) Client {
+	return &SSHClient{
+		Host:                  host,
+		User:                  user,
+		Password:              password,
+		PKPath:                pkPath,
+		InsecureIgnoreHostKey: insecureIgnoreHostKey,
+	}
+}
+
+/*
+
+ */
+
+type SSHClientMock struct{}
```

Comment:
Is this supposed to be in this file?

Reply:
Initially, I wanted to have mock implementation to make testing easier. But since its possible to setup containers, this shouldn't be necessary. I removed the mock code (also in other code locations). 
Good catch!